src/serprog.c

```diff
@@ -0,0 +1,510 @@
+/*
+ * avrdude - A Downloader/Uploader for AVR device programmers
+ * Support for using serprog programmers to program over ISP.
+ * For information on serprog see:
+ * https://flashrom.org/supported_hw/supported_prog/serprog/index.html
+ *
+ * Copyright (C) 2024 Sydney Louisa Wilke <git@funkeleinhorn.com>
+ * used linuxspi.c as a template:
```

Comment:
Good practice to cite what you based your work on. In this case I don't think it's needed (and perhaps even a bit misleading) as Cuzner/Ramsauer will have relied on templates before them and they most likely would not have generated the templates themselves. Can be removed, but if you feel strongly, leave it in.

Reply:
Perhaps move the reference to linuxspi.c to specific functions where you based your coding on theirs, eg, `_enable()`